pkg/cmd/certregenerationcontroller/cmd.go

```diff
@@ -0,0 +1,169 @@
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/version"
+)
+
+type Options struct {
+	controllerContext *controllercmd.ControllerContext
+
+	TLSServerName string
+}
+
+func NewCertRegenerationControllerCommand(ctx context.Context) *cobra.Command {
+	o := &Options{
+		TLSServerName: "localhost-recovery",
+	}
+
+	cmd := controllercmd.
+		NewControllerCommandConfig("cert-regeneration-controller", version.Get(), func(ctx context.Context, controllerContext *controllercmd.ControllerContext) error {
+			o.controllerContext = controllerContext
+
+			err := o.Validate(ctx)
+			if err != nil {
+				return err
+			}
+
+			err = o.Complete(ctx)
+			if err != nil {
+				return err
+			}
+
+			err = o.Run(ctx)
+			if err != nil {
+				return err
+			}
+
+			return nil
+		}).NewCommandWithContext(ctx)
+	cmd.Use = "cert-regeneration-controller"
+	cmd.Short = "Start the Cluster Certificate Regeneration Controller"
+
+	cmd.PersistentFlags().StringVarP(&o.TLSServerName, "tls-server-name", "", o.TLSServerName, "The SNI hostname to set for the server in kubeconfig")
+
+	return cmd
+}
+
+func (o *Options) Validate(ctx context.Context) error {
+	return nil
+}
+
+func (o *Options) Complete(ctx context.Context) error {
+	return nil
+}
+
+func (o *Options) Run(ctx context.Context) error {
+	if len(o.TLSServerName) != 0 {
+		// TLSServerName chooses the SNI serving endpoint on the apiserver.
+		// Particularly useful when connecting to "localhost" and wanting to choose a special
+		// serving endpoint like "localhost-recovery" that has long-lived serving certs
+		// for localhost connections.
+		o.controllerContext.KubeConfig.TLSClientConfig.ServerName = o.TLSServerName
+		o.controllerContext.ProtoKubeConfig.TLSClientConfig.ServerName = o.TLSServerName
+	}
+
+	kubeClient, err := kubernetes.NewForConfig(o.controllerContext.ProtoKubeConfig)
+	if err != nil {
+		return fmt.Errorf("can't build kubernetes client: %w", err)
+	}
+
+	configClient, err := configeversionedclient.NewForConfig(o.controllerContext.KubeConfig)
+	if err != nil {
+		return fmt.Errorf("failed to create config client: %w", err)
+	}
+
+	configInformers := configexternalinformers.NewSharedInformerFactory(configClient, 10*time.Minute)
+
+	kubeAPIServerInformersForNamespaces := v1helpers.NewKubeInformersForNamespaces(
+		kubeClient,
+		metav1.NamespaceAll,
+		operatorclient.GlobalMachineSpecifiedConfigNamespace,
+		operatorclient.GlobalUserSpecifiedConfigNamespace,
+		operatorclient.OperatorNamespace,
+		operatorclient.TargetNamespace,
+	)
+
+	operatorClient, dynamicInformers, err := genericoperatorclient.NewStaticPodOperatorClient(o.controllerContext.KubeConfig, operatorv1.GroupVersion.WithResource("kubeapiservers"))
+	if err != nil {
+		return err
+	}
+
+	eventRecorder := events.NewKubeRecorder(kubeClient.CoreV1().Events(""), "openshift-kube-apiserver-operator", &corev1.ObjectReference{
+		APIVersion: "v1",
+		Kind:       "namespace",
+		Name:       "openshift-kube-apiserver-operator",
```

Comment:
no, put a name unique the this binary in here.

Reply:
> no, put a name unique the this binary in here.

on line 113, where you're passing a name.

Reply:
wired it for the builder created recorder